internal/tcptunnel/tcptunnel.go

```diff
@@ -160,15 +160,18 @@ func (tun *Tunnel) run(ctx context.Context, local io.ReadWriter, rawJWT string)
 		http.StatusFound,
 		http.StatusTemporaryRedirect,
 		http.StatusPermanentRedirect:
-		if rawJWT == "" {
+		if retryCount == 0 {
 			_ = remote.Close()
 
-			authURL, err := url.Parse(res.Header.Get("Location"))
```

Comment:
If my understanding is correct, we can't use the returned redirect URL because it won't have the programmatic login endpoint.

An alternative solution would be to add support for programmatic login to the authenticate service but I'm not sure there's any advantage to that.

Reply:
> If my understanding is correct, we can't use the returned redirect URL because it won't have the programmatic login endpoint.

Yes.

> An alternative solution would be to add support for programmatic login to the authenticate service but I'm not sure there's any advantage to that.

For now this makes sense. I think the way we are currently handling programmatic auth flow has some legacy logic where the audience claim is tied to the proxy's domain (which is set from the proxy handler endpoint). This made more sense when we had client-side maintained state, but makes less sense now where I think we don't really care about the upstream audience.